tensilelite/Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -43,6 +45,130 @@
 
 namespace Tensile
 {
+    namespace streamk
+    {
+        namespace math
+        {
+            /**
+             * Performs `(n + d - 1) / d`, but is robust against the case where
+             * `(n + d - 1)` would overflow.
+             */
+            template <typename N, typename D>
+            __device__ __host__ inline constexpr N safe_ceil_div(N n, D d)
```

Comment:
Might be able to use `ceilDivide`: https://github.com/ROCm/hipBLASLt/blob/fecd96c2ecb4064aa3c11610207b565605061ecd/tensilelite/Tensile/TensileInstructions/Utils.py#L195C5-L195C15, `safe_ceil_div` is probably better version of the implementation.